sphinx_gallery/gen_rst.py

```diff
@@ -654,3 +657,45 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         logger.debug("%s ran in : %.2g seconds", src_file, time_elapsed)
 
     return intro, time_elapsed
+
+
+def gen_binder_url(fname, binder_conf):
+    if not isinstance(binder_conf, dict) or len(binder_conf) == 0:
+        return None
+    req_values = ['url', 'org', 'repo', 'branch', 'dependencies']
+
+    # Ensure all fields are populated
+    missing_values = [val for val in req_values
+                      if not isinstance(binder_conf.get(val, None),
+                                        (str, unicode))]
+    if len(missing_values) > 0:
+        raise ValueError('binder_conf is missing values for: {}'.format(
+            missing_values))
+
+    # Ensure we have a requirements file
+    allowed_reqs_files = ['requirements.txt', 'environment.yml']
+    path_reqs = binder_conf['dependencies']
+    if not any(path_reqs.endswith(ii) for ii in allowed_reqs_files):
+        raise ValueError('Requirements file must be one of {}'.format(
+            allowed_reqs_files))
```

Comment:
L675-685, has nothing to do with getting the url. Make this an isolated function. Also, in `gen_gallery.py` you check for the configuration to be present to copy the requirements. Here you check the configuration to be present, and complete and for the requirements file to be present, and you do it for every example. Check all this exception raising stuff early in the configuration step. Then you can be sure it is already correct by now, and only generate the `url`, which is the purpose of the function.